src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs

```diff
@@ -43,32 +50,62 @@ internal class FunctionDispatcher : IFunctionDispatcher
         {
             _metricsLogger = metricsLogger;
             _scriptOptions = scriptHostOptions.Value;
+            _environment = environment;
             _languageWorkerChannelManager = languageWorkerChannelManager;
             _eventManager = eventManager;
             _workerConfigs = languageWorkerOptions.Value.WorkerConfigs;
-            _logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryFunctionDispatcher);
+            _managedDependencyOptions = managedDependencyOptions;
+            _logger = loggerFactory.CreateLogger<FunctionDispatcher>();
+            _workerRuntime = _environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName);
```

Comment:
Could this be an empty string if all functions indexed are written in the same language?

Reply:
workerRuntime is inferred from the content if environment variable is not set. Please see

https://github.com/Azure/azure-functions-host/blob/350631e01cebcc6df8deff6fa1ffa24c40304d84/src/WebJobs.Script/Host/ScriptHost.cs#L275

Call to initialize function dispatcher included inferred workerRuntime